rust/schema/annotations.py

```diff
@@ -0,0 +1,1741 @@
+
+
+@annotate(YeetExpr)
+class _:
+    """
+    A `yeet` expression. For example:
+    ```rust
+    if x < size {
+       do yeet "index out of bounds";
+    }
+    ```
+    """
+
+
+@annotate(RecordExprField)
+class _:
+    """
+    A field in a record expression. For example `a: 1` in:
+    ```rust
+    Foo { a: 1, b: 2 };
+    ```
+    """
+
+
+@annotate(RecordExpr)
+class _:
+    """
+    A record expression. For example:
+    ```rust
+    let first = Foo { a: 1, b: 2 };
+    let second = Foo { a: 2, ..first };
+    Foo { a: 1, b: 2 }[2] = 10;
+    Foo { .. } = second;
+    ```
+    """
+
+
+@annotate(FieldExpr)
+class _:
+    """
+    A field access expression. For example:
+    ```rust
+    x.foo
+    ```
+    """
+
+
+@annotate(AwaitExpr)
+class _:
+    """
+    An `await` expression. For example:
+    ```rust
+    async {
+        let x = foo().await;
+        x
+    }
+    ```
+    """
+
+
+@annotate(CastExpr)
+class _:
+    """
+    A cast expression. For example:
+    ```rust
+    value as u64;
+    ```
+    """
+
+
+@annotate(RefExpr)
+class _:
+    """
+    A reference expression. For example:
+    ```rust
+        let ref_const = &foo;
+        let ref_mut = &mut foo;
+        let raw_const: &mut i32 = &raw const foo;
+        let raw_mut: &mut i32 = &raw mut foo;
+    ```
+    """
+
+
+@annotate(PrefixExpr)
+class _:
+    """
+    A unary operation expression. For example:
+    ```rust
+    let x = -42
+    let y = !true
+    let z = *ptr
+    ```
+    """
+
+
+@annotate(BinaryExpr)
+class _:
+    """
+    A binary operation expression. For example:
+    ```rust
+    x + y;
+    x && y;
+    x <= y;
+    x = y;
+    x += y;
+    ```
+    """
+
+
+@annotate(RangeExpr)
+class _:
+    """
+    A range expression. For example:
+    ```rust
+    let x = 1..=10;
+    let x = 1..10;
+    let x = 10..;
+    let x = ..10;
+    let x = ..=10;
+    let x = ..;
+    ```
+    """
+
+
+@annotate(IndexExpr)
+class _:
+    """
+    An index expression. For example:
+    ```rust
+    list[42];
+    list[42] = 1;
+    ```
+    """
+
+
+@annotate(ClosureExpr)
+class _:
+    """
+    A closure expression. For example:
+    ```rust
+    |x| x + 1;
+    move |x: i32| -> i32 { x + 1 };
+    async |x: i32, y| x + y;
+     #[coroutine]
+    |x| yield x;
+     #[coroutine]
+     static |x| yield x;
+    ```
+    """
+
+
+@annotate(TupleExpr)
+class _:
+    """
+    A tuple expression. For example:
+    ```rust
+    (1, "one");
+    (2, "two")[0] = 3;
+    ```
+    """
+
+
+@annotate(ArrayExpr)
+class _:
+    """
+    An array expression. For example:
+    ```rust
+    [1, 2, 3];
+    [1; 10];
+    ```
+    """
+    pass
+
+
+# @annotate(ElementListExpr)
+# class _:
+#     """
+#     An element list expression. For example:
+#     ```rust
+#     [1, 2, 3, 4, 5];
+#     [1, 2, 3, 4, 5][0] = 6;
+#     ```
+#     """
+
+
+# @annotate(RepeatExpr)
+# class _:
+#     """
+#     A repeat expression. For example:
+#     ```rust
+#     [1; 10];
+#     ```
+#     """
```

Comment:
I guess this is what because of what you mentioned on https://github.com/github/codeql-team/issues/3378, right?

Reply:
Yes, but I can remove this for now.